app/services/work_time_saver.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+class WorkTimeSaver
+  attr_reader :work_time
+
+  def initialize(work_time)
+    @work_time = work_time
+    @old_payload = @work_time.integration_payload
+  end
+
+  def call(additional_params = {})
+    external_auth = work_time.project&.external_auth
+    if external_auth.nil? || work_time.task.nil?
+      return save_without_payload(additional_params)
+    end
+
+    if (payload = validate_integration(work_time, external_auth))
+      save_with_payload(payload, additional_params)
+    else
+      InvalidWorkTime.new(work_time)
+    end
+  end
+
+  private
+
+  class InvalidWorkTime
```

Comment:
let's try to solve this in some other way i.e creating from object class with validations which will copy validation errors from WorkTime.

```
class WorkTimeForm
  include ActiveModel::Model
  validate :validates_ticker_url

  def save
    ...
  end

  def valid?
    super
    copy_errors
    errors.empty?
  end

  private

  def copy_errors
    work_time.valid?
    work_time.errors.details.each do |key, value|
      errors.add(key, value.first[:error])
    end
  end
end
```

Reply:
Changed